coordinator-service/src/blob-chunk-storage.ts

```diff
@@ -93,10 +91,8 @@ export class BlobChunkStorage implements ChunkStorage {
         const destinationClient = this.containerClient.getBlobClient(
             destinationBlobName,
         )
-        const abortSignal = AbortController.timeout(timeoutMilliseconds)
-        const result = await destinationClient.syncCopyFromURL(sourceUrl, {
-            abortSignal,
-        })
+        const poller = await destinationClient.beginCopyFromURL(sourceUrl)
+        const result = await poller.pollUntilDone()
```

Comment:
My original implementation wasn't super well thought out but:

I didn't use this initially because I thought an explicit timeout would be useful for handling / managing HTTP connections.  "block (and hold the HTTP connection open) until the copy is complete" _feels_ a little scary.

Also, many frontends / load balancers / HTTP routers, like an nginx ingress controller, are going to have a connection timeout (e.g., 60 seconds).  Having the upstream system forcefully close the connection can lead to confusing behavior, so I'd rather timeout here and log a message.

Can you increase the maximum timeout instead?

Reply:
Apparently `beginCopyFromURL` also receives `abortSignal`. That solves the issue, right?

Reply:
BTW, I think `beginCopyFromURL` addresses your worry differently anyway - you launch the operation remotely and then send polling requests.